### DISCUSSION TITLE
How can I print the Wifi name and password on the serial screen ?

### DISCUSSION THREAD

**Discussion by 10furkan60** [Hieromon/AutoConnect#313] [Q&A] [answered]
Firstly thank you so much for this library. For my school project, I need to show the wifi name and password, but I don't know how to get it from EEPROM after connecting to WiFi

Thank all to help
![0](https://user-images.githubusercontent.com/77199607/108038165-92a3d280-704b-11eb-9df6-be40de896690.JPG)

**Comment by Hieromon** [comment, marked as answer]
For ESP32, you can get the SSID and Password of the current AP after a connection has been established by calling the API of the ESP-IDF natively.

```cpp
wifi_config_t conf;
if (esp_wifi_get_config(WIFI_IF_STA, &conf) == ESP_OK) {
  Serial.printf("SSID: %s\n", (char*)conf.sta.ssid);
  Serial.printf("Password: %s\n", (char*)conf.sta.password);
}
```
- [esp_wifi_get_config(wifi_interface_t, wifi_config_t*)](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp_wifi.html#_CPPv419esp_wifi_get_config16wifi_interface_tP13wifi_config_t)
  Get [the configuration](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/wifi.html#station-basic-configuration) of specified interface. You should specify `WIFI_IF_STA` for the 1st parameter. The current configuration is returned into the second parameter.

- [wifi_sta_config_t](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp_wifi.html#_CPPv417wifi_sta_config_t)
  STA configuration settings for the ESP32. It has an SSID and Password as public members.

**Reply by 10furkan60** [reply]
Thank u so much bro ))
